internal/k8s/k8s_test.go

```diff
@@ -0,0 +1,54 @@
+package k8s
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeServer "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestClient_CreateAPIKeySecret(t *testing.T) {
+
+	t.Run("successful creation", func(tt *testing.T) {
+
+		secretNamespace := "tink-system"
+		secretName := "colony-api"
+		secretKey := "api-key"
+		secretValue := "super-duper-secret"
+
+		mockServer := fakeServer.NewClientset()
```

Comment:
Nitpick: In Go, multiple declarations can be wrapped in this format:

```suggestion
var (
	secretNamespace = "tink-system"
	secretName      = "colony-api"
	secretKey       = "api-key"
	secretValue     = "super-duper-secret"

	mockServer = fakeServer.NewClientset()
)
```

Makes it nice to read 😄